### PR TITLE
1991 empty state v1 for tables

### PR DESCRIPTION
**Pull request by martins5225**
This introduces an empty state feature to data tables for both standard and custom objects. 

The new EmptyTable component displays a user-friendly message and an optional action button when the table is empty,

### CODE REVIEW

**Comment by lucasbordeau** [comment]
Hi @martins5225, thanks for your proposal. This seems neat, could you clean up a little bit the files, and post some screenshot before/after so that @Bonapara can review the UX/UI ? Also is it linked to an existing issue ? If that's the case, could you link it ?

File: front/src/modules/ui/data/data-table/components/DataTable.tsx [hunk 1]
```diff
@@ -118,8 +127,20 @@ export const DataTable = ({ updateEntityMutation }: DataTableProps) => {
     },
   });
 
+  const tableRowIds = useRecoilValue(tableRowIdsState);
+  // const visibleTableColumns = useRecoilScopedValue(
```
Comment (lucasbordeau) [inline front/src/modules/ui/data/data-table/components/DataTable.tsx:131]: Remove dead code

File: front/src/modules/ui/data/data-table/components/DataTable.tsx [hunk 2]
```diff
@@ -118,8 +127,20 @@ export const DataTable = ({ updateEntityMutation }: DataTableProps) => {
...
+  //   visibleTableColumnsScopedSelector,
+  //   TableRecoilScopeContext,
+  // );
+
+  const defaultOnClick = () => {
```
Comment (lucasbordeau) [inline front/src/modules/ui/data/data-table/components/DataTable.tsx:136]: Prefer allowing undefined in EmptyTable onClick method. And check if onClick is defined in EmptyTable

File: front/src/modules/ui/data/data-table/components/EmptyTable 2.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import styled from '@emotion/styled';
```
Comment (lucasbordeau) [inline front/src/modules/ui/data/data-table/components/EmptyTable 2.tsx:1]: Is this a duplicate ?

**Review by lucasbordeau — CHANGES_REQUESTED** [review]

**Comment by martins5225** [comment]
> Hi @martins5225, thanks for your proposal. This seems neat, could you clean up a little bit the files, and post some screenshot before/after so that @Bonapara can review the UX/UI ? Also is it linked to an existing issue ? If that's the case, could you link it ?

HI @lucasbordeau Thank you for the feedback, I'll do the clean up and also include the screenshots. I don't think this is linked to an existing issue

**Comment by Bonapara** [comment]
Hi @martins5225 Can you even provide a **screenshot video** of the feature? Can you make sure the content is well centered (No scroll should be possible) and that it doesn't **blink** before the content loading when there is 1+ record(s)?

**Comment by charlesBochet** [comment]
@martins5225 I'm closing this PR due to inactivity. Feel free to re-open it with new changes and we will happily take another look :) Thanks for the help anyway!

**Pull request closed without merging**